docs/config/processors/componentsData.js

```diff
@@ -5,7 +5,17 @@ var buildConfig = require('../../../config/build.config.js');
 // add many kilobytes of loading overhead.
 function publicDocData(doc, extraData) {
   var options = _.assign(extraData || {}, { hasDemo: (doc.docType === 'directive') });
-  return buildDocData(doc, options, 'components');
+
+  // This RegEx always retrieves the last source descriptor.
+  // For example it retrieves from `/opt/material/src/core/services/ripple/ripple.js` the following
+  // source descriptor: `src/core/`.
+  // This is needed because components are not only located in `src/components`.
+  var descriptor = doc.fileInfo.filePath.toString().match(/src\/.*?\//g).pop();
```

Comment:
I feel like the node `fs` api might be able to do this more easily, but I'm not that familiar with it.
@robertmesserle  ?